api/common/upgradeseries.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
```

Comment:
The methods in this file are still tested in api/uniter/unit_test.go

Reply:
Why can we not move them. Is there a gotcha in moving them to an `upgradeseries_test.go` file.

Reply:
Moving them involves pulling over the juju conn suite style of testing.  From what I understand, it does a good job in the Uniter, but is fragile and not something we wish to propagate more.

Most of the common api pieces seem to be tested within the api's they're embedded in.  However I added a api/common/upgradeseries_test.go with pieces

Reply:
Acknowledged